File: TopInterview150/Intervals/MinNumArrowsToBurstBalloons/solution.py
class Solution(object):
    def findMinArrowShots(self, points):
        """
        :type points: List[List[int]]
        :rtype: int
        """
        points.sort(key = lambda x: x[1])
        counter = 1
        shotPointer = points[0][1]

        for point in points:
            if point[0] > shotPointer:
                counter+=1
                shotPointer = point[1]

        return counter

File: TopInterview150/Intervals/MinNumArrowsToBurstBalloons/solution.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](const vector<int>& a, const vector<int>& b)
        {
            return a.back() < b.back();
        });
        int counter = 1;
        int leftmostEnd = points[0][1];
        for (vector<int> point : points)
        {
            if (point[0] > leftmostEnd)
            {
                counter++;
                leftmostEnd = point[1];   
            }
        }

        return counter;
    }
};